### ISSUE TITLE
Bug: Saving selected videos is not working

### ISSUE THREAD

**Issue by mikkovedru** [telegramdesktop/tdesktop#28803]
### Steps to reproduce

1. Right-click on video and choose "Save as...". It works fine.
2. Right-click on video and choose "Select"
3. Right-click on the same video again and choose "Download Selected".
4. (BUG) It gives the directory selecting dialog, but the saving itself does not work.

This is especially frustrating when trying to select a bunch of media (pics and videos), saving them, seeing that files appeared, but not noticing that only pics were saved, but not videos.

### Expected behaviour

All the selected media should be saved.

### Actual behaviour

They are not saved.

### Operating system

Linux Mint 21.3 Cinnamon

### Version of Telegram Desktop

5.9 (latest)

### Installation source

Static binary from official website

**Comment by ilya-fedin** [comment]
> 4\. (BUG) It gives the directory selecting dialog, but the saving itself does not work.

I can't reproduce, it just starts downloading after I click "Download Selected". Perhaps some steps to reproduce are missing (e.g. settings customization)?

**Comment by mikkovedru** [comment]
Good hunch!

My settings: `Advanced` -> `Ask download path for each file` was on.

When I turned it off and chose the directory, selecting the file and commanding to download selected was working.

But when I turned the same setting on again, I got this error:
![2024-12-28_](https://github.com/user-attachments/assets/52bad096-dcf4-4991-bf98-89054534c37b)

When I turned the setting on/off and re-tried again, the downloading of the selected file failed in the original report's fashion - without any indication/error message.

**Comment by ilya-fedin** [comment]
No, I still can't reproduce. Given that you get such a dialog, perhaps your system file dialog really returns an invalid download location? At this point I believe you better reporting this to the relevant system component.

**Comment by mikkovedru** [comment]
> perhaps your system file dialog really returns an invalid download location?

I don't think that is probable. It's not like the system file dialog gives a separate path each time the file is being saved (if Telegram does request directory info (which it already has) with each file of the batch, then this is a clear inefficiency and should be fixed).

Instead, the system file dialog provides the path one time only, after which Telegram tries to save all the selected files into the same directory. And Telegram does it successfully with the image files, proving that there is no inherent problem on the OS side. Saving images works, but saving videos (even if it is one selected video) does not. All the logic tells that the issue must be internal to Telegram, in the processing difference between images and videos.

Is there something I can do to help you pinpoint the issue? Do some testing, try different Telegram versions, etc? I am willing to help.

**Comment by ilya-fedin** [comment]
> It's not like the system file dialog gives a separate path each time the file is being saved

That could happen. I saw that in the past. Given that you use a distro with 2 years old software, it's highly likely you're still using xdg-desktop-portal version with that bug.

> Is there something I can do to help you pinpoint the issue?

Try to update to linux mint 22. I believe the issue will go away.

**Comment by mikkovedru** [comment]
> Given that you use a distro with 2 years old software

FYI Linux Mint 21.3 Cinnamon was released in Jan of this year and is supported until April 2027.

My system has `xdg-desktop-portal` of the version 1.14.4. The [latest version](https://github.com/flatpak/xdg-desktop-portal) is 1.18.4 released on 18.4.2024.


> it's highly likely you're still using xdg-desktop-portal version with **that** bug

What do you mean by "that bug"? Do you have any link to the specific bug in mind, or were you just talking in theoretically and in general?


> Try to update to linux mint 22. I believe the issue will go away.

ok. Once I try I will inform you about the results.

**Comment by ilya-fedin** [comment]
> FYI Linux Mint 21.3 Cinnamon was released in Jan of this year and is supported until April 2027.

Yet, it has 2 years old software as it's based on Ubuntu 22.04

> My system has `xdg-desktop-portal` of the version 1.14.4. The [latest version](https://github.com/flatpak/xdg-desktop-portal) is 1.18.4 released on 18.4.2024.

Your version is 2 years old
![](https://github.com/user-attachments/assets/a0d63c12-9599-464a-b64c-3833bd76a1a5)

> What do you mean by "that bug"?

I mean there were lots of reports that file dialog works wrongly 2 years ago, including that the portal returns a path to /var/run instead of the actual one. I don't have a link to upstream report, sorry, I only know tdesktop was bombarded by issues like that back then. Only time made them irrelevant.

**Comment by mikkovedru** [comment]
Thank you for the info!
